subprojects/base-services/src/main/java/org/gradle/internal/concurrent/ManagedExecutorImpl.java

```diff
@@ -88,23 +88,21 @@ public void stop(int timeoutValue, TimeUnit timeoutUnits) throws IllegalStateExc
         executorPolicy.onStop();
     }
 
-    public void shutdown() {
-        executor.shutdown();
-    }
-
-    public List<Runnable> shutdownNow() {
-        return executor.shutdownNow();
-    }
-
-    public boolean isShutdown() {
-        return executor.isShutdown();
-    }
-
-    public boolean isTerminated() {
-        return executor.isTerminated();
-    }
-
-    public boolean awaitTermination(long timeout, TimeUnit unit) throws InterruptedException {
-        return executor.awaitTermination(timeout, unit);
+    @Override
+    public void setFixedPoolSize(int numThreads) {
+        if (executor instanceof ThreadPoolExecutor) {
+            ThreadPoolExecutor threadPoolExecutor = (ThreadPoolExecutor) executor;
+            // we do this because in jdk9 these methods throw an exception if you
```

Comment:
I just didn't want someone "simplifying" this to one specific order because it worked on a jdk <9.  I've captured this in an follow-up issue to address these comments.